File: balanced_binary_tree.c
#include <math.h>
#include <stdlib.h>
#include <stdbool.h>
/**
 * Definition for a binary tree node.
 */
struct TreeNode {
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
};

#define max(x, y) ( (x) > (y) ? (x) : (y) )

int getHeight(struct TreeNode *root) {
    int height = 0;
    if (root)
        height = 1 + max(getHeight(root->left), getHeight(root->right));
    return height;
}

bool isBalanced(struct TreeNode* root) {
    if (root) {
        if (isBalanced(root->left) == false)
            return false;
        if (isBalanced(root->right) == false)
            return false;
        if (abs(getHeight(root->left) - getHeight(root->right)) > 1)
            return false;
    }
    return true;
}

int main (void) {
    return 0;
}

File: py/tree/BalancedBinaryTree.py
# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution(object):
    def isBalancedHelper(self, root):
        if root == None:
            return True, 0
        boolLeft, highLeft = self.isBalancedHelper(root.left)
        boolRight, highRight = self.isBalancedHelper(root.right)
        b = (boolLeft and boolRight and abs(highLeft - highRight) <= 1)
        high = 1 + max(highLeft, highRight)
        return b, high        
        
    def isBalanced(self, root):
        if root == None:
            return True
        b, high = self.isBalancedHelper(root)
        return b
